junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertIterableEquals.java

```diff
@@ -24,38 +24,38 @@
 import java.util.function.Supplier;
 
 /**
- * {@code AssertIterable} is a collection of utility methods that support asserting
- * Iterable equality in tests.
+ * {@code AssertIterable} is a collection of utility methods that support
+ * asserting Iterable equality in tests.
  *
  * @since 5.0
  */
 class AssertIterableEquals {
 
-	///CLOVER:OFF
+	/// CLOVER:OFF
 	private AssertIterableEquals() {
 		/* no-op */
 	}
-	///CLOVER:ON
+	/// CLOVER:ON
 
 	static void assertIterableEquals(Iterable<?> expected, Iterable<?> actual) {
-		assertIterableEquals(expected, actual, () -> null);
+		assertIterableEquals(expected, actual, (String) null);
 	}
 
 	static void assertIterableEquals(Iterable<?> expected, Iterable<?> actual, String message) {
-		assertIterableEquals(expected, actual, () -> message);
+		assertIterableEquals(expected, actual, new ArrayDeque<>(), message);
 	}
 
 	static void assertIterableEquals(Iterable<?> expected, Iterable<?> actual, Supplier<String> messageSupplier) {
 		assertIterableEquals(expected, actual, new ArrayDeque<>(), messageSupplier);
 	}
 
 	private static void assertIterableEquals(Iterable<?> expected, Iterable<?> actual, Deque<Integer> indexes,
-			Supplier<String> messageSupplier) {
+			Object messageContainer) {
```

Comment:
Please undo **all** renamings of `messageSupplier` to `messageContainer`.

Reply:
It's not just renaming but changing the parameter. I've done this in several methods. All they except of one are `private`. I would prefer to keep this change in order to avoid massive code duplication in the API module.

Reply:
OK.

I'll have to review it in detail on my machine.